### PR TITLE
Refactor emails

### PR DESCRIPTION
**Pull request by gregorykan**
**DiscussionMailer**

- [x] convert `DiscussionEmailInfo` into `DiscussionEmailHelper`, stateless methods
- [x] refactor `DiscussionMailer`
  - [x] comment_replied_to
  - [x] discussion_announced
  - [x] discussion_edited (no tests, unused)
  - [x] invitation_created
  - [x] new_comment
  - [x] new_discussion (no tests, unused)
  - [x] user_mentioned
- [x] refactor discussion emails so that they don't depend on instance variables
- [x] setup e2e suite for the discussion emails

**PollMailer**

- [x] convert `PollEmailInfo` into `PollEmailHelper`, stateless methods
- [x] refactor `PollMailer`
- [x] refactor poll emails so that they don't depend on instance variables
  - [x] invitation_created (unused, no tests)
  - [x] invitation_resend (unused, no tests)
  - [x] outcome_announced
  - [x] poll_closing_soon
  - [x] poll_closing_soon_author
  - [x] poll_edited
  - [x] poll_option_added (unused, see poll_option_added_author/no tests)
  - [x] poll_option_added_author
  - [x] user_mentioned
  - [x] user_reminded (unused?, no tests)
  - [x] outcome_created (unused, see outcome_announced/no tests)
  - [x] outcome_created_author (unused, no tests)
  - [x] poll_announced
  - [x] poll_expired (unused, no tests)
  - [x] poll_expired_author
  - [x] poll_closed (unused, see: poll_expired/no tests)
  - [x] poll_created (unused, see poll_announced/no tests)
  - [x] stance_created
- [x] setup e2e suite for (selected) poll emails - 61 tests, 100% coverage of used templates (see above)

### CODE REVIEW

File: app/mailers/discussion_mailer.rb [hunk 1]
```diff
@@ -8,32 +11,32 @@ class DiscussionMailer < BaseMailer
...
-    send_single_mail  to: @info.recipient.email,
-                      from: from_user_via_loomio(@info.actor),
-                      reply_to: reply_to_address_with_group_name(model: @info.eventable, user: @info.recipient),
-                      subject_key: event.email_subject_key || "discussion_mailer.#{action_name}.subject",
-                      subject_params: { actor: @info.actor.name,
-                                        group: @info.discussion.group.full_name,
-                                        discussion: @info.discussion.title },
-                      locale: @info.recipient.locale
+    send_single_mail  to: @recipient.email,
+                      from: from_user_via_loomio(@event.user),
```
Comment (robguthrie) [inline app/mailers/discussion_mailer.rb]: do we need to anonymous screen this?

File: app/views/discussion_mailer/_footer.html.haml [hunk 2]
```diff
@@ -1,12 +1,12 @@
 .thread-mailer__footer
-  - if @info.recipient.is_logged_in?
-    %img.thread-mailer__footer-image{src: @info.pixel_src, alt: '', width: 1, height: 1}
+  - if recipient.is_logged_in?
+    %img.thread-mailer__footer-image{src: pixel_src(event, recipient), alt: '', width: 1, height: 1}
```
Comment (robguthrie) [inline app/views/discussion_mailer/_footer.html.haml]: can we verify that this works?

File: app/views/discussion_mailer/_header.html.haml [hunk 3]
```diff
@@ -1,4 +1,4 @@
 %table.thread-mailer__subject
   %tr
-    %td= render 'base_mailer/avatar', user: @info.actor
-    %td= t(:"discussion_mailer.#{@info.action_name}.subject_html", actor: @info.actor.name, discussion: link_to(@info.discussion.title, @info.target_url))
+    %td= render 'base_mailer/avatar', user: event.user
```
Comment (robguthrie) [inline app/views/discussion_mailer/_header.html.haml]: anonymous_or_actor_for(event)

File: app/views/poll_mailer/common/_footer.html.haml [hunk 4]
```diff
@@ -1,7 +1,6 @@
 .poll-mailer-common-footer
   .poll-mailer-common-footer__spacer= PollMailer::REPLY_DELIMITER
   .poll-mailer-common-footer__text
-    -# %div= t(:"poll_mailer.common.why_send_this", reason: @info.send_reason)
-    = link_to t(:"poll_mailer.common.view_on_loomio", site_name: AppConfig.theme[:site_name]), @info.target_url
+    = link_to t(:"poll_mailer.common.view_on_loomio", site_name: AppConfig.theme[:site_name]), target_url({}, poll, recipient)
```
Comment (robguthrie) [inline app/views/poll_mailer/common/_footer.html.haml]: maybe convert target_url to have optional args last... or all args are named params: eventable, recipient, args

**Comment by robguthrie** [comment]
If you decide this is ready for deploy, I think it should be in a window of time where we can rollback then deploy new builds (probably just fixing a missing var, typo etc), and when you will personally be around to respond. So maybe tomorrow Tuesday or Monday next week?

File: app/views/discussion_mailer/_footer.html.haml (hunk 2 above)
Comment (gregorykan) [reply app/views/discussion_mailer/_footer.html.haml]: i notice that the specs have been commented out, and we left some comments. let's pick this up separately

**Pull request merged**